File: java/src/string_manipulation/LC_1957_DeleteCharactersToMakeFancyString.java
package string_manipulation;
// https://leetcode.com/problems/delete-characters-to-make-fancy-string

public class LC_1957_DeleteCharactersToMakeFancyString {
    public String makeFancyString(String s) {
        char[] ch = s.toCharArray();
        int counter = 1, index = 1;
        for (int i = 1; i < ch.length; i++) {
            if (ch[i] != ch[i - 1]) {
                counter = 1;
            } else {
                counter += 1;
            }
            if (counter > 2)
                ch[index++] = ch[i];
        }

        return new String(ch, 0, index);
    }
}

File: python/string_manipulation/LC_1957_DeleteCharactersToMakeFancyString.py
# https://leetcode.com/problems/delete-characters-to-make-fancy-string

class Solution:
    def makeFancyString(self, s: str) -> str:
        counter = 1
        output = [s[0]]
        for i in range(1, len(s)):
            if s[i] != s[i-1]:
                counter = 1
            else:
                counter += 1
            if counter < 3:
                output.append(s[i])

        return "".join(output)
